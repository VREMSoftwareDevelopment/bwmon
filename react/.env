VITE_VERSION=$npm_package_version
VITE_NAME=$npm_package_name
VITE_DESCRIPTION=$npm_package_description
