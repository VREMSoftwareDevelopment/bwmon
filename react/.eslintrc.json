{
    "extends": [
        "eslint:recommended",
        "plugin:import/errors",
        "plugin:react/recommended",
        "plugin:jsx-a11y/recommended",
        "plugin:prettier/recommended"
    ],
    "plugins": ["react", "import", "jsx-a11y", "jest", "prettier"],
    "settings": {
        "react": {
            "version": "detect"
        }
    },
    "rules": {
        "react/prop-types": "off",
        "indent": ["error", 4],
        "linebreak-style": "error",
        "quotes": ["error", "single"],
        "no-unused-vars": [
            "error",
            {
                "vars": "all",
                "varsIgnorePattern": "^_",
                "args": "after-used",
                "argsIgnorePattern": "^_",
                "caughtErrors": "all",
                "caughtErrorsIgnorePattern": "^_"
            }
        ]
    },
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "env": {
        "es6": true,
        "browser": true,
        "node": true,
        "jest/globals": true
    }
}
