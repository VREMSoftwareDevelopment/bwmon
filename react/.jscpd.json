{
    "minLines": 10,
    "minTokens": 50,
    "reporters": ["console", "html"],
    "output": "reports/duplication",
    "ignore": ["**/*.test.js", "**/*.test.jsx"]
}
