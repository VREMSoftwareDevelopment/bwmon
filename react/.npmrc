save=true
puppeteer_skip_download=true
PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true