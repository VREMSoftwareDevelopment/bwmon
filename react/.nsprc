{
    "1070415": {
        "active": true,
        "expiry": "2022-12-31",
        "notes": "Ignored - waiting for update"
    }
}
