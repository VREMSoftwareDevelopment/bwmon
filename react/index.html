<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <link rel="icon" href="/favicon.ico" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
        <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
        <meta name="theme-color" content="#000000" />
        <meta name="Bandwidth Monitor" content="Bandwidth MonitorApplication" />
        <link rel="manifest" href="/manifest.json" />
        <base href="/" />
        <title>Bandwidth Monitor</title>
    </head>
    <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root"></div>
        <script type="module" src="/src/index.jsx"></script>
    </body>
</html>
