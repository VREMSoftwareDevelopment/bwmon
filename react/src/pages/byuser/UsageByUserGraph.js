/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { makeStyles } from '@mui/styles';
import { Paper, Grid2 } from '@mui/material';
import DropDown from '../../components/inputs/DropDown';
import Search from '../../components/inputs/Search';
import useUsageByUserGraph from '../../hooks/byuser/UseUsageByUserGraph';
import Loading from '../../components/loading/Loading';
import Graph from '../../components/graph/Graph';

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        margin: theme.spacing(2),
        marginTop: 6,
    },
}));

const UsageByUserGraph = () => {
    const classes = useStyles();

    const { options, series, years, year, setYear, months, month, setMonth, setFilter, loading } = useUsageByUserGraph();

    const handleChangeYear = (event) => setYear(event.target.value);

    const handleChangeMonth = (event) => setMonth(event.target.value);

    const handleChangeFilter = (event) => setFilter(event.target.value);

    const itemYear = () => (
        <Grid2 item sm={2}>
            <DropDown data-testid="user-year-graph" id="user-year-graph" onChange={handleChangeYear} items={years} value={year} />
        </Grid2>
    );

    const itemMonth = () => (
        <Grid2 item sm={2}>
            <DropDown
                data-testid="user-month-graph"
                id="user-month-graph"
                onChange={handleChangeMonth}
                items={months}
                value={month}
            />
        </Grid2>
    );

    const itemUser = () => (
        <Grid2 item sm={4}>
            <Search data-testid="user-filter-graph" id="user-filter-graph" onChange={handleChangeFilter} />
        </Grid2>
    );

    const items = () => (
        <div className={classes.root}>
            <Grid2 container spacing={4}>
                {itemYear()}
                {itemMonth()}
                {itemUser()}
            </Grid2>
        </div>
    );

    return (
        <Paper>
            <Loading isLoading={loading} />
            {items()}
            <Graph options={options} series={series} />
        </Paper>
    );
};

export default UsageByUserGraph;
