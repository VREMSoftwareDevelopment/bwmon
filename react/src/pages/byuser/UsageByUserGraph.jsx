/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { Paper } from '@mui/material';
import Grid from '@mui/material/Grid2';
import DropDown from '../../components/inputs/DropDown';
import Search from '../../components/inputs/Search';
import useUsageByUserGraph from '../../hooks/byuser/UseUsageByUserGraph';
import Loading from '../../components/loading/Loading';
import Graph from '../../components/graph/Graph';
import { graphStyles } from '../../utils/StylesUtils';

const UsageByUserGraph = () => {
    const classes = graphStyles();

    const { options, series, years, year, setYear, months, month, setMonth, setFilter, loading } = useUsageByUserGraph();

    const handleChangeYear = (event) => setYear(event.target.value);

    const handleChangeMonth = (event) => setMonth(event.target.value);

    const handleChangeFilter = (event) => setFilter(event.target.value);

    return (
        <Paper>
            <Loading isLoading={loading} />
            <div className={classes.root}>
                <Grid container spacing={4}>
                    <Grid sm={2}>
                        <DropDown
                            data-testid="user-year-graph"
                            id="user-year-graph"
                            onChange={handleChangeYear}
                            items={years}
                            value={year}
                        />
                    </Grid>
                    <Grid sm={2}>
                        <DropDown
                            data-testid="user-month-graph"
                            id="user-month-graph"
                            onChange={handleChangeMonth}
                            items={months}
                            value={month}
                        />
                    </Grid>
                    <Grid sm={4}>
                        <Search data-testid="user-filter-graph" id="user-filter-graph" onChange={handleChangeFilter} />
                    </Grid>
                </Grid>
            </div>
            <Graph options={options} series={series} />
        </Paper>
    );
};

export default UsageByUserGraph;
