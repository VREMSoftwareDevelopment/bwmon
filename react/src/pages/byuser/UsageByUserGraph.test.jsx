/*
 *      Copyright (C) 2010 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { render, screen, within, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import UsageByUserGraph from './UsageByUserGraph';
import useUsageByUserGraph from '../../hooks/byuser/UseUsageByUserGraph';
import Search from '../../components/inputs/Search';

jest.mock('../../components/graph/Graph');
jest.mock('../../components/inputs/Search');
jest.mock('../../hooks/byuser/UseUsageByUserGraph');

describe('UsageByUserGraph', () => {
    const data = {
        options: {},
        series: [],
        years: [2020, 2021, 2022],
        year: 2021,
        setYear: jest.fn(),
        months: ['January', 'February', 'March'],
        month: 'February',
        setMonth: jest.fn(),
        setFilter: jest.fn(),
        loading: false,
    };

    beforeEach(() => {
        useUsageByUserGraph.mockReturnValue(data);
    });

    const theme = createTheme();

    const renderComponent = (props) =>
        render(
            <ThemeProvider theme={theme}>
                <UsageByUserGraph {...props} />
            </ThemeProvider>
        );

    it('renders loading state', () => {
        useUsageByUserGraph.mockReturnValue({ ...data, loading: true });
        renderComponent();
        expect(screen.getByText('Loading...')).toBeInTheDocument();
    });

    it('renders graph when data is loaded', () => {
        renderComponent();
        expect(screen.getByTestId('test-graph-id')).toBeInTheDocument();
    });

    it('renders year selector', () => {
        renderComponent();
        const container = screen.getByTestId('user-year-graph');
        expect(container).toBeInTheDocument();
        const { getByText } = within(container);
        expect(getByText(2021)).toBeInTheDocument();
    });

    it('handles year selector', () => {
        renderComponent();
        const container = document.querySelector('#user-year-graph');
        fireEvent.mouseDown(container);
        fireEvent.click(screen.getByRole('option', { name: 2020 }));
        expect(useUsageByUserGraph().setYear).toHaveBeenCalledWith(2020);
    });

    it('renders month selector', () => {
        renderComponent();
        const container = screen.getByTestId('user-month-graph');
        expect(container).toBeInTheDocument();
        const { getByText } = within(container);
        expect(getByText('February')).toBeInTheDocument();
    });

    it('handles month selector', () => {
        renderComponent();
        const container = document.querySelector('#user-month-graph');
        fireEvent.mouseDown(container);
        fireEvent.click(screen.getByRole('option', { name: 'January' }));
        expect(useUsageByUserGraph().setMonth).toHaveBeenCalledWith('January');
    });

    it('displays correct graph options and series', () => {
        useUsageByUserGraph.mockReturnValue({
            options: { chart: { id: 'test-chart' } },
            series: [{ name: 'test-series', data: [1, 2, 3] }],
            loading: false,
        });
        renderComponent();
        expect(screen.getByTestId('test-graph-id')).toBeInTheDocument();
        expect(screen.getByTestId('test-graph-id')).toHaveTextContent('Graph');
        expect(screen.getByTestId('test-graph-id')).toHaveTextContent('{"chart":{"id":"test-chart"}}');
        expect(screen.getByTestId('test-graph-id')).toHaveTextContent('[{"name":"test-series","data":[1,2,3]}]');
    });

    it('handles user filter change', () => {
        Search.mockImplementation(({ onChange }) => <input data-testid="user-filter-graph" onChange={onChange} />);
        renderComponent();
        fireEvent.change(screen.getByTestId('user-filter-graph'), { target: { value: 'John' } });
        expect(data.setFilter).toHaveBeenCalledWith('John');
    });
});
