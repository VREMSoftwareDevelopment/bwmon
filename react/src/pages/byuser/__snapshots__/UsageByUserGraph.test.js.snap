// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageByUserGraph renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="makeStyles-root-2"
  >
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <div
      className="makeStyles-root-3"
    >
      <div
        className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
        role="alert"
      >
        <div
          className="MuiAlert-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          className="MuiAlert-message"
        >
          <div
            className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-root-1"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
              disabled={false}
              id="user-filter"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="IP, MAC or USER"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
      />
    </div>
  </div>
  <div>
    Graph
    <p>
      Options: 
      {}
    </p>
    <p>
      Series: 
      []
    </p>
  </div>
</div>
`;
